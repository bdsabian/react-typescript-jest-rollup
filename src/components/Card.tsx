import * as React from 'react';

export interface CardProps {
  title: string;
  image: string;
  body: string;
}

export const Card = ({ title, image, body }: CardProps) => (
  <div className="card">
    <div className="title">{title}</div>
    <div className="image">
      <img src={image} />
    </div>
    <div className="body">{body}</div>
  </div>
);

export default Card;
