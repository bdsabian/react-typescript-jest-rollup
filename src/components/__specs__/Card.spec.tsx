import * as React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';

describe('Card', () => {
  const wrapper = shallow(<Card title="Test" image="https://i.imgur.com/GIXGUZC.jpg" body="This is a card" />);

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
