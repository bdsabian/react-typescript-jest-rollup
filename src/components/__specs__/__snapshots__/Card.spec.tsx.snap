// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="card"
>
  <div
    className="title"
  >
    Test
  </div>
  <div
    className="image"
  >
    <img
      src="https://i.imgur.com/GIXGUZC.jpg"
    />
  </div>
  <div
    className="body"
  >
    This is a card
  </div>
</div>
`;
